lib/matplotlib/backend_bases.py

```diff
@@ -138,6 +145,35 @@ def get_registered_canvas_class(format):
     return backend_class
 
 
+class MainLoopBase(object):
```

Comment:
Never used ABC before, happy to try, but not sure of the benefits here over duck-typing.